forte/data/data_store.py

```diff
@@ -474,3 +476,79 @@ def prev_entry(self, tid: int) -> List:
             The previous entry of the same type as the ``tid`` entry.
         """
         raise NotImplementedError
+
+    def parse_onto_file(self):
+        r"""If user provides a customized ontology json file, we parse
+        this file and set self._type_attributes accordingly. It not only
+        includes the classes in the provided ontology file, but also import
+        classes from parent entry files.
+        """
+        if self.onto_file_path is None:
+            return
+
+        ontoGen = OntologyCodeGenerator()
+        merged_scheme, merged_dict = [], []
+        ontoGen.parse_ontology_spec(
+            self.onto_file_path, merged_scheme, merged_dict
+        )
+
+        self._type_attributes = {}
+
+        for onto in merged_scheme:
+            entry_name = onto["entry_name"]
+            attr_dic = {}
+            idx = 4
+            if "attributes" in onto:
+                for attr_dict in onto["attributes"]:
+                    name = attr_dict["name"]
+                    attr_dic[name] = idx
+                    idx += 1
+            entry_dic = {}
+            entry_dic["parent_entry"] = onto["parent_entry"]
+            entry_dic["attributes"] = attr_dic
+            self._type_attributes[entry_name] = entry_dic
+
+    def _get_entry_attributes_by_class(
+        self, input_entry_class_name: str
+    ) -> List:
+        """Get type attributes by class name. `input_entry_class_name` should be
+        a fully qualified name of an entry class.
+
+        The `dataclass` module<https://docs.python.org/3/library/dataclasses.html> can add
+        generated special methods to user-defined classes. There is an in-built function
+        called `__dataclass_fields__` that is called on the class object, and it returns
```

Comment:
```suggestion
        called `__dataclass_fields__` that is called on the class object, and it returns
```

This sentence reads weird to me, maybe rephrase?

Reply:
not fixed: https://github.com/asyml/forte/blob/dfd9c2ad1e31d9b4670967a3b67964f35b574372/forte/data/data_store.py#L991

Reply:
this is still there